Sources/Models/API/Product.swift

```diff
@@ -158,6 +159,26 @@ struct Product: Mappable {
     var minerals: [Tag]?
     var nucleotides: [Tag]?
     var otherNutrients: [Tag]?
+    private var _productAttributes: ProductAttributes?
+    var productAttributes: ProductAttributes? {
+        //validate barcode before retrieving
+        get {
+            if _productAttributes?.barcode == barcode {
+                return _productAttributes
+            } else {
+                return nil
+            }
+        }
+
+        //validate barcode before setting
+        set {
+            if let productAttr = newValue, productAttr.barcode == self.barcode {
+                self._productAttributes = productAttr
+            } else {
+                return
+            }
+        }
+    }
```

Comment:
Using a middle-man property to safely get product attributes and set them only if the barcodes match.